compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -201,6 +223,7 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
             sym.isClass // reference to this in inline methods
           )
     case None =>
+      (sym.hasAnnotation(defn.InternalQuoted_QuoteTypeTagAnnot) && level > 0) ||
       sym.is(Package) || sym.owner.isStaticOwner || levelOK(sym.owner)
```

Comment:
Minor performance tuning: switch the two lines?

Maybe add a comment for the added line.